### PR TITLE
improve sending error notifications

### LINKED ISSUE #2635: “Sending cancelled“ notifications are not dismissable
When I cancel a image uploading, the sending notification becomes "sending cancelled" but it's not dismissable nor clickable and I have to force stop tusky to get rid of the notification. 

* * * *
- Tusky Version: 19.0
- Android Version: 12
- Android Device: Pixel 6 Pro
- Mastodon instance (if applicable): pixelfed.social

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by connyduck**
Clicking the error or cancel notification will now open drafts, both can be dismissed and the cancel one auto-cancels after 5 sec. Not sure about the auto cancelling, but it was the original behavior that got broken in some refactoring.

Closes #2635

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/service/SendStatusService.kt [hunk 1]
```diff
@@ -259,7 +266,41 @@ class SendStatusService : Service(), Injectable {
...
-        return PendingIntent.getService(this, statusId, intent, NotificationHelper.pendingIntentFlags(false))
+        return PendingIntent.getService(
+            this,
+            statusId,
+            intent,
+            NotificationHelper.pendingIntentFlags(false)
+        )
+    }
+
+    private fun draftNotification(
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/service/SendStatusService.kt:277]: I would rename it to `showDraftNotification()`

**Review by charlag — APPROVED** [review]
I think cancelling error is really misleading, otherwise should be fine?

File: app/src/main/java/com/keylesspalace/tusky/service/SendStatusService.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/service/SendStatusService.kt:277]: it is not showing, just building it though

**Comment by connyduck** [comment]
> I think cancelling error is really misleading, otherwise should be fine?

what exactly is misleading?

**Comment by charlag** [comment]
> > I think cancelling error is really misleading, otherwise should be fine?
> 
> what exactly is misleading?

You send a post, throw the phone in your pocket, it fails and shows notifications for 5s, you never see that it fails. Or does it not work like that? I might be confused

**Comment by connyduck** [comment]
> > > I think cancelling error is really misleading, otherwise should be fine?
> > 
> > 
> > what exactly is misleading?
> 
> You send a post, throw the phone in your pocket, it fails and shows notifications for 5s, you never see that it fails. Or does it not work like that? I might be confused

The error notification stays forever until user clicks or dismisses. If you click cancel on an ongoing sending notification, it turns into a cancel notification that stays only for 5 seconds.

**Comment by charlag** [comment]
Ah, sorry, I see now, it's good!

**Pull request merged**